packages/analyticsdx-template-lint/src/linter.ts

```diff
@@ -611,15 +613,19 @@ export abstract class TemplateLinter<
     templateType: 'app' | 'embeddedapp',
     templateTypeNode: JsonNode | undefined
   ): Promise<void> {
-    // for app templates, it needs to have at least 1 dashboard, dataflow, externalFile, lens, or recipe specified,
-    // so accumulate the total of each (handling the -1 meaning no node) and the property nodes for each
-    // empty array field
+    // for app templates, it needs to have at least 1 ones of these specified, so accumulate the total of each
+    // (handling the -1 meaning no node) and the property nodes for each empty array field
     const { count, nodes } = [
+      { data: lengthJsonArrayAttributeValue(tree, 'components'), name: 'components' },
```

Comment:
I noticed that this check was stale in vscode and that server actually checks for any 1 of these so I updated this (and the tests).